### PR TITLE
Chore: Update Detox to 20.1.2

### PR DESCRIPTION
**Pull request by diegolmello**
## Proposed changes

Updates Detox and makes it run on CI.

## Screenshots
image

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

[NATIVE-72]

[NATIVE-72]: https://rocketchat.atlassian.net/browse/NATIVE-72?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# DON'T COMMIT THIS FILE
```
Comment (GleidsonDaniel) [inline .env:1]: 👀

File: e2e/data.ts [hunk 2]
```diff
@@ -54,48 +26,15 @@ const data = {
...
-		username: `newuserfour${value}`,
-		password: `passwordfour${value}`,
-		email: `mobile+registeringfour${value}@rocket.chat`
-	},
-	random: value
+	randomUser: (): { username: string; name: string; password: string; email: string } => {
+		const randomVal = random();
+		return {
+			username: `user${randomVal}`,
+			name: `user${randomVal}`, // FIXME: apply a different name
```
Comment (GleidsonDaniel) [inline e2e/data.ts:33]: Will this stay for the next version?

**Review by GleidsonDaniel — APPROVED** [review]
I tested it on android and ios and it worked normally, before it didn't work, that for me is already a big advance hahaha

Just these two comments there, I think it's safe to merge.

File: .env (hunk 1 above)
Comment (diegolmello) [reply .env:1]: hahahaha after my commit :)
it's used during tests.

File: e2e/data.ts (hunk 2 above)
Comment (diegolmello) [reply e2e/data.ts:33]: Yes, we need to fix a few components first.
I didn't want to do it on this PR, because it's already too big.

**Pull request merged**